Exec/science/xrb_spherical/problem_tagging.H

```diff
@@ -25,19 +26,24 @@ void problem_tagging(int i, int j, int k,
     GpuArray<Real, 3> loc;
     position(i, j, k, geomdata, loc);
 
-    if (problem::tag_by_density) {
-        if (state(i,j,k,URHO) > problem::cutoff_density &&
-            state(i,j,k,UFS) / state(i,j,k,URHO) > problem::X_min) {
+    if (problem::tag_by_thermodynamics) {
 
-            Real dist = std::abs(loc[0]);
+        // Consider refining when density and temp are above some threshold
+        if (state(i,j,k,URHO)  > problem::cutoff_density &&
+            state(i,j,k,UTEMP) > problem::cutoff_temp) {
 
-            if (level < problem::max_hse_tagging_level && dist < geomdata.ProbLo(0) + problem::r_refine_distance) {
+            if (level < problem::max_base_tagging_level) {
                 tag(i,j,k) = TagBox::SET;
             }
-        }
 
-        if (state(i,j,k,URHO) > problem::cutoff_density) {
-            if (level < problem::max_base_tagging_level) {
+            // find the distance in cylindrical radial direction
+            Real dist = std::abs(loc[0] * std::sin(loc[1]));
+
+            // Allow higher level refinement to better resolve
+            // the initial burning phase which is at the pole of the NS.
+            if (dist < problem::refine_distance &&
```

Comment:
you can avoid a divide by putting density on the other side of the <

Reply:
done